alpa/pipeline_parallel/apply_grad.py

```diff
@@ -302,29 +316,28 @@ def _get_apply_grad_outvar_constraints(jax_pipeline_stages, stage_to_mesh,
 
 def process_apply_gradient(apply_grad_jaxpr, barrier, acc_grad_dict,
                            jax_pipeline_stages, stage_to_mesh, gensym_func,
-                           num_micro_batches, num_meshes, global_invars,
-                           global_outvars, donated_invars):
+                           num_meshes, global_invars, global_outvars,
+                           donated_invars):
     """Slice apply_grad jaxpr into stages and assign them to the correspondig meshes."""
     # TODO(yonghao): the condition of creating RDA variable should be extended.
 
     # Process apply gradient:
     # 1. change invars of apply grad to outvars of accumulate grad
     gradients = [g for g in barrier.outvars if not isinstance(g, DropVar)]
+    assert len(gradients) == len(barrier.invars)
     mask = {
```

Comment:
Not related to this PR. Can you change the name of this variable from `mask` to something meaningful? For example `barrier_out2in`